plugins/security/nexus-ldap-client/src/main/java/org/sonatype/security/realms/ldap/client/internal/JerseyConfiguration.java

```diff
@@ -0,0 +1,289 @@
+import org.sonatype.security.realms.ldap.api.dto.LdapServerConfigurationDTO;
+import org.sonatype.security.realms.ldap.api.dto.LdapServerRequest;
+import org.sonatype.security.realms.ldap.api.dto.LdapUserAndGroupAuthConfigurationDTO;
+import org.sonatype.security.realms.ldap.client.Configuration;
+import org.sonatype.security.realms.ldap.client.Connection;
+import org.sonatype.security.realms.ldap.client.Connection.Host;
+import org.sonatype.security.realms.ldap.client.Connection.Protocol;
+import org.sonatype.security.realms.ldap.client.Mapping;
+import org.sonatype.security.realms.ldap.client.internal.JerseyConfiguration.ConfigurationXO;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Strings;
+import com.sun.jersey.api.client.ClientHandlerException;
+import com.sun.jersey.api.client.UniformInterfaceException;
+import org.apache.commons.codec.binary.Base64;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * An LDAP configurations.
+ */
+public class JerseyConfiguration
+    extends EntitySupport<Configuration, ConfigurationXO>
+    implements Configuration
+{
+  /**
+   * A hack to not expose Restlet ancient DTOs but keep state. The restlet DTOs are hidden and used only
+   * for transport, basically keeping client users "safe" from changing them.
+   */
+  // TODO: currently XO has getters/setters to enable BeanUtils simple copy, see EntitySupport
+  public static class ConfigurationXO
+  {
+    private String name;
+
+    private Connection connection;
+
+    private Mapping mapping;
+
+    public String getName() {
+      return name;
+    }
+
+    public void setName(final String name) {
+      this.name = name;
+    }
+
+    public Connection getConnection() {
+      return connection;
+    }
+
+    public void setConnection(final Connection connection) {
+      this.connection = connection;
+    }
+
+    public Mapping getMapping() {
+      return mapping;
+    }
+
+    public void setMapping(final Mapping mapping) {
+      this.mapping = mapping;
+    }
+  }
+
+  private final JerseyNexusClient client;
+
+  public JerseyConfiguration(final JerseyNexusClient client) {
+    super(null);
+    this.client = client;
+  }
+
+  public JerseyConfiguration(final JerseyNexusClient client, final LdapServerConfigurationDTO settings) {
+    super(settings.getId(), toXO(settings));
+    this.client = client;
+  }
+
+  @Override
+  public String getName() {
+    return settings().name;
+  }
+
+  @Override
+  public Configuration setName(final String name) {
+    checkNotNull(name);
+    settings().name = name;
+    return this;
+  }
+
+  @Override
+  public Connection getConnection() {
+    return settings().connection;
+  }
+
+  @Override
+  public Mapping getMapping() {
+    return settings().mapping;
+  }
+
+  @Override
+  protected ConfigurationXO createSettings(final String id) {
+    final ConfigurationXO data = new ConfigurationXO();
+    data.name = null;
```

Comment:
This method is invoked from ctor but also from refresh too, where a reload from server happens. I just found it to be more expressive by nulling it explicitly.